### ISSUE TITLE
Improve depth learning + evaluation

### ISSUE THREAD

**Issue by vict0rsch** [cc-ai/omnigan#133]
https://arxiv.org/pdf/2003.06620.pdf

lets use this issue's comments to debrief: interesting stuff, questions etc.

**Comment by melisandeteng** [comment]
What we can keep in mind from the some models presented in this overview: 
The authors explicitly discourage use of the depth prediction model on higher resolution than the one is was trained on. 
Also, it seems a lot of them just take NYU depth dataset, or KITTI dataset to benchmark on, but in our case it's very important to have a model that's robust across different types of images. 
Hence, we'll focus on [this model MiDaS](https://github.com/intel-isl/MiDaS) for the moment

**Comment by melisandeteng** [comment]
We implemented the loss with gradient matching  + scale invariant MSE terms. 

just to keep this reference somewhere: [DADA](https://github.com/valeoai/DADA/blob/master/dada/utils/func.py) uses the reverse Huber loss

**Comment by melisandeteng** [comment]
architectures : 
[Megadepth](https://github.com/zhengqili/MegaDepth/blob/master/pytorch_DIW_scratch.py) : Hourglass network from [this paper](https://arxiv.org/pdf/1604.03901.pdf) - composed of modified inception modules
[MiDaS](https://github.com/intel-isl/MiDaS/blob/master/midas/midas_net.py) : resnet-based architecture from [this paper](https://openaccess.thecvf.com/content_cvpr_2018/papers/Xian_Monocular_Relative_Depth_CVPR_2018_paper.pdf)
